third_party/rust/url/v2/crate/src/lib.rs

```diff
@@ -0,0 +1,404 @@
+// Copyright 2013-2015 The rust-url developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::cmp;
+use std::fmt::{self, Write};
+use std::hash;
+use std::str;
+use std::error::Error;
+use std::convert::TryFrom;
+use std::fmt::Formatter;
+
+macro_rules! simple_enum_error {
+    ($($name: ident => $description: expr,)+) => {
+        /// Errors that can occur during parsing.
+        ///
+        /// This may be extended in the future so exhaustive matching is
+        /// discouraged with an unused variant.
+        #[derive(PartialEq, Eq, Clone, Copy, Debug)]
+        #[non_exhaustive]
+        pub enum ParseError {
+            $(
+                $name,
+            )+
+        }
+
+        impl fmt::Display for ParseError {
+            fn fmt(&self, fmt: &mut Formatter<'_>) -> fmt::Result {
+                match *self {
+                    $(
+                        ParseError::$name => fmt.write_str($description),
+                    )+
+                }
+            }
+        }
+    }
+}
+
+impl Error for ParseError {}
+
+simple_enum_error! {
```

Comment:
nit: you could use `thiserror` instead if you want to get rid of the macro

Reply:
not looking to create more dependencies here :)